pkg/prometheus/server/operator.go

```diff
@@ -1098,7 +1100,19 @@ func (c *Operator) updateConfigResourcesStatus(ctx context.Context, p *monitorin
 	}); err != nil {
 		return fmt.Errorf("listing all ServiceMonitors from cache failed: %w", err)
 	}
-	return getErr
+	if getErr != nil {
+		return getErr
+	}
+
+	// Update selected podMonitors status.
```

Comment:
I'd rather move after L1067 (e.g. update selected smons, pmons, ... and then remove out of date bindings).